pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -1003,14 +1017,23 @@ private synchronized TaskDebugInfo getTaskDebugInfo(WorkflowContext workflowCont
       TaskCount subtaskCount = new TaskCount();
       for (int partition : partitionSet) {
         // First get the partition's state and update the subtaskCount
+        String taskIdForPartition = jobContext.getTaskIdForPartition(partition);
         TaskPartitionState partitionState = jobContext.getPartitionState(partition);
+        TaskConfig helixTaskConfig = jobConfig.getTaskConfig(taskIdForPartition);
+        PinotTaskConfig pinotTaskConfig = null;
+        if (helixTaskConfig != null) {
+          pinotTaskConfig = PinotTaskConfig.fromHelixTaskConfig(helixTaskConfig);
+          if ((tableNameWithType != null) && (!tableNameWithType.equals(pinotTaskConfig.getTableName()))) {
+            // Filter task configs that match this table name
+            continue;
+          }
+        }
```

Comment:
moving this logic before recording the task state to avoid recording the entries that are skipped when table name is specified